File: greedy/leetcode_3216/solution.js
/**
 * @param {string} s
 * @return {string}
 */
var getSmallestString = function(s) {
    let result = s;
    const arr = s.split('');

    const checkSameParity = (num1, num2) => {
        return Number(num1) % 2 === Number(num2) % 2
    }

    for (let i=0; i<arr.length - 1; i++) {
        const isSameParity = checkSameParity(arr[i], arr[i+1]);

        if(isSameParity && arr[i] > arr[i+1]) {
            const currentArr = [...arr];

            currentArr[i] = arr[i+1];
            currentArr[i+1] = arr[i];

            const currentString = currentArr.join("");

            if(currentString < result) {
                result = currentString
            }
        }
    }

    return result;
};

File: greedy/leetcode_3216/solution.py
class Solution:
    def getSmallestString(self, s: str) -> str:
        result = s
        arr = list(s)

        def check_same_parity(num_1: str, num_2:str) -> bool:
            return int(num_1) % 2 == int(num_2) % 2

        for i in range(len(arr) - 1):
            is_same_parity = check_same_parity(arr[i], arr[i+1])

            if is_same_parity and arr[i] > arr[i+1]:
                current_arr = arr[:]

                current_arr[i] = arr[i+1]
                current_arr[i+1] = arr[i]

                current_string = "".join(current_arr)

                if current_string < result:
                    result = current_string

        return result
